umcontroller/umcontroller.go

```diff
@@ -400,6 +403,14 @@ func (umCtrl *Controller) UpdateComponents(
 	return umCtrl.currentComponents, umCtrl.updateError
 }
 
+func (umCtrl *Controller) SubscribeNewComponentListener() <-chan cloudprotocol.ComponentStatus {
+	listener := make(chan cloudprotocol.ComponentStatus)
```

Comment:
Channel size should be a  least 1. Otherwise, channel will be blocked if there is no channel reader.

Reply:
fixed